pkg/relayer/cmd/cmd.go

```diff
@@ -198,7 +206,29 @@ func setupRelayerDependencies(
 		config.NewSupplyQueryClientContextFn(queryNodeGRPCUrl),            // leaf
 		config.NewSupplyTxClientContextFn(queryNodeGRPCUrl, txNodeRPCUrl), // leaf
 		config.NewSupplyDelegationClientFn(),                              // leaf
-		config.NewSupplySharedQueryClientFn(),                             // leaf
+
+		// Setup the params caches and configure them to clear on new blocks.
+		// TODO_TECHDEBT: Consider a flag to change client queriers caching behavior.
```

Comment:
I'm going to push on the fact that this is a TODO_IN_THIS_PR.

It's not that hard and I want to understand the benefit (if any) of this cache.

It feels like we've built a car but not actually checking if it works.